### ISSUE TITLE
System Catalog Tables

### ISSUE THREAD

**Issue by ahsanhadi** [pgsql-io/pgosql#2]
System catalog tables and views are created to define the structure of the database, they are also referred as data dictionary.   This ticket is create to list down some of the system catalog tables available in Oracle and are widely used. It is always advisable to create user defined views using the system catalog tables since the system catalog table structure might, however it doesn't happen very often.

All the system catalog tables will be created in SYS schema,  the SYS and SYSTEM default users and schema are defined in issue#1.

SYS.DUAL

DUAL is one of the most used system table in Oracle, it is single row and single column table that is used to return a single row from a Select statement that doesn't reference any other table. It is also used to execute functions using select statement. It is also very useful to use the DUAL table for returning a static value or computing a constant expression. 

Column name : Dummy
Type : VARCHAR(1)
Not Null

SYS.TABLES

The TABLES table stores information about the tables in the database including the name, owner, number of columns, size row etc. The TABLES table also contain information about the system tables.

Detailed description will be added.

SYS.INDEXES

The INDEXES table stores information about the table indexes in the database.

Detailed description will be added.

SYS.SEQUENCES

The SEQUENCES table stores information about the sequences in the database.

Detailed description will be added.

SYS.USER_DEPENDENCIES

The USER_DEPENDENCIES table stores information use dependencies in the database, this contains information of dependencies between user objects.

Detailed description will be added.

**Comment by luss** [comment]
Yup, we need all of this.  My initial thoughts are that we have our own
fork of OraFCE that is SSPL licensed and adds the needful to the SYS schema.
